### PR TITLE
tweaks

### PR DESCRIPTION
**Pull request by SamVerschueren**
- fix `.travis.yml`
- ES2015ify readme
- Add API in readme
- Use `m` instead of `dotProp` in test file

@sindresorhus Can you check the descriptions and types in the API?

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -40,6 +40,32 @@ console.log(obj);
...
+
+#### path
+
+Type: `string`
+
+Path of the property in the object.
+
+#### value
+
+Type: `*`
```
Comment (SamVerschueren) [inline readme.md]: Not sure about this one. Value can be anything right?

Comment (kevva) [inline readme.md]: Yeah.

File: readme.md [hunk 2]
```diff
@@ -40,6 +40,32 @@ console.log(obj);
...
+
+Type: `object`
+
+Object to get or set the path value.
+
+#### path
+
+Type: `string`
+
+Path of the property in the object.
```
Comment (stevemao) [inline readme.md]: This should state it supports `.` path and escaping the `.` by `\\.`

File: readme.md (hunk 1 above)
Comment (stevemao) [inline readme.md]: Type: `any`?

File: readme.md [hunk 3]
```diff
@@ -40,6 +40,32 @@ console.log(obj);
...
+
+### get(obj, path)
+
+### set(obj, path, value)
+
+#### obj
+
+Type: `object`
+
+Object to get or set the path value.
```
Comment (stevemao) [inline readme.md]: I think this needs to be reworded.

File: test.js [hunk 4]
```diff
@@ -1,26 +1,26 @@
 import test from 'ava';
-import dotProp from './';
+import m from './';
```
Comment (stevemao) [inline test.js]: Is this the naming convention in `AVA`?

**Comment by stevemao** [comment]
Should we add links (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
http://docs.mongodb.org/manual/core/document/#dot-notation) just to let people know why this notation?

File: test.js (hunk 4 above)
Comment (SamVerschueren) [inline test.js]: It's the new @sindresorhus convention ;). At first, it was `fn`, but `fn` isn't general enough.

File: readme.md (hunk 3 above)
Comment (SamVerschueren) [inline readme.md]: Feel free to do some suggestions :).

Comment (stevemao) [inline readme.md]: I'd say something like:
Object that the `get` or `set` operation work on

My English is bad though :disappointed:

File: test.js (hunk 4 above)
Comment (sindresorhus) [inline test.js]: Discussion: https://github.com/sindresorhus/generator-nm/issues/37

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: What does Flow and TypeScript use?

File: readme.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ dotProp.get({foo: {'dot.dot': 'unicorn'}}, 'foo.dot\\.dot');
 //=> 'unicorn'
 
 // setter
-var obj = {foo: {bar: 'a'}};
+let obj = {foo: {bar: 'a'}};
```
Comment (sindresorhus) [inline readme.md]: Shouldn't this be `const`? You're not changing the binding of `obj`, just its contents.

File: .travis.yml [hunk 6]
```diff
@@ -1,5 +1,5 @@
-sudo: false
```
Comment (sindresorhus) [inline .travis.yml]: You can't remove this on existing repos. Only on new ones. That's how Travis deal with the transition.

File: readme.md (hunk 5 above)
Comment (SamVerschueren) [inline readme.md]: Good catch :)

File: readme.md (hunk 1 above)
Comment (SamVerschueren) [inline readme.md]: TypeScript uses `any`.

File: .travis.yml (hunk 6 above)
Comment (SamVerschueren) [inline .travis.yml]: Oh boy... Thanks for tip!

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: :+1:

File: readme.md (hunk 3 above)
Comment (sindresorhus) [inline readme.md]: I think the current description is fine. Maybe add \` around it so it's clear the path is the argument shown above.

File: readme.md (hunk 2 above)
Comment (SamVerschueren) [inline readme.md]: @sindresorhus Do you mean it's ok as is, or do you want me to rephrase that with the dot property stuff?

Something like?

> Path of the property in the object. Use `.` for nested objects or `\\.` to add a `.` in a key.

Comment (sindresorhus) [inline readme.md]: > Path of the property in the object. Use . for nested objects or \\. to add a . in a key.

:+1:

**Comment by SamVerschueren** [comment]
Processed all the feedback.

@stevemao Don't really feel the need to add those links. It's pretty basic stuff, right? :)

**Comment by sindresorhus** [comment]
Thanks @SamVerschueren :)

I don't really see the point of adding those links either.

**Comment by stevemao** [comment]
Thanks @SamVerschueren :smile:

**Pull request merged**